src/main/java/com/walking/intensive/chapter2/task6/Task6.java

```diff
@@ -44,7 +62,18 @@ static int getGcd(int m, int n) {
      * <p>Если входные данные некорректны - метод должен возвращать -1.
      */
     static int getGcdByEuclideanAlgorithm(int m, int n) {
-        // Ваш код
-        return 0;
+
+        if (m <= 0 || n <= 0) {
+            return -1;
+        }
+
+        int minValue = Math.min(m, n);
+        int modValue = Math.max(m, n) % minValue;
+
+        if (modValue == 0) {
+            return minValue;
+        }
+
+        return getGcdByEuclideanAlgorithm( minValue, modValue );
```

Comment:
Error: eckstyle] [ERROR] /home/runner/work/intensive-tasks-2024/intensive-tasks-2024/src/main/java/com/walking/intensive/chapter2/task6/Task6.java:77:42: '(' is followed by whitespace. [ParenPad]
Error: eckstyle] [ERROR] /home/runner/work/intensive-tasks-2024/intensive-tasks-2024/src/main/java/com/walking/intensive/chapter2/task6/Task6.java:77:63: ')' is preceded with whitespace. [ParenPad]